go/trace/trace.go

```diff
@@ -97,17 +101,31 @@ func RegisterSpanFactory(sf SpanFactory) {
 	spanFactory = sf
 }
 
-type fakeSpanFactory struct{}
-
-func (fakeSpanFactory) New(parent Span) Span                                         { return fakeSpan{} }
-func (fakeSpanFactory) FromContext(ctx context.Context) (Span, bool)                 { return nil, false }
-func (fakeSpanFactory) NewContext(parent context.Context, span Span) context.Context { return parent }
+var spanFactory SpanFactory = fakeSpanFactory{}
 
-// fakeSpan implements Span with no-op methods.
-type fakeSpan struct{}
+var (
+  tracingServer = flag.String("tracer", "noop", "tracing service to use. available are: noop, jaeger. Configuration is provided using environment variables.")
+)
 
-func (fakeSpan) StartLocal(string)            {}
-func (fakeSpan) StartClient(string)           {}
-func (fakeSpan) StartServer(string)           {}
-func (fakeSpan) Finish()                      {}
-func (fakeSpan) Annotate(string, interface{}) {}
+// StartTracing enables tracing for a named service
+func StartTracing(serviceName string) io.Closer {
+  switch *tracingServer {
```

Comment:
alternatively, we could use `package init` to register from each tracingService, and avoid the need to explicitly change the following code. 

```
var tracingServices map[string]func()io.Closer

// in fake.go
func init(){
 tracingServices['fake'] = func(){return &nilCloser{}}
}

// in jaeger.go
func init(){
 tracingServices['jaeger']=func(){
    tracer, closer, err := NewJagerTracerFromEnv(serviceName)
    if err != nil {
      log.Error(vterrors.Wrapf(err, "failed to create a jaeger tracer"))
      return &nilCloser{}
    }
    // Register it for all openTracing enabled plugins, mainly the grpc connections
    opentracing.SetGlobalTracer(tracer)

     // Register it for the internal Vitess tracing system
    RegisterSpanFactory(OpenTracingFactory{Tracer: tracer})
    return closer
 }
}
```

so that for a new tracing service, like `zipkin`, i could add just a `zipkin.go` that has a similar `init`

Reply:
this is exciting! thank u so much for taking this closer to life! i was wondering if zipkin could be seamlessly integrated with `opentracing api`, but then i've found: https://github.com/openzipkin-contrib/zipkin-go-opentracing

so i guess i'm happy with the choice at least, wish i could speak for the more general audience 🤣

Reply:
Neat. Yeah, your `init()` solution looks way better. I took it and changed it a bit but it's the same idea. Thanks!